pkg/serviceaccount/metrics.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serviceaccount
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const kubeServiceAccountSubsystem = "serviceaccount"
+
+var (
+	// LegacyTokensTotal is the number of legacy tokens used against apiserver.
+	LegacyTokensTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      kubeServiceAccountSubsystem,
+			Name:           "legacy_tokens_total",
+			Help:           "Cumulative legacy service account tokens used",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// StaleTokensTotal is the number of stale projected tokens not refreshed on
+	// client side.
+	StaleTokensTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      kubeServiceAccountSubsystem,
+			Name:           "stale_tokens_total",
+			Help:           "Cumulative stale projected service account tokens used",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// ValidTokensTotal is the number of valid projected tokens used.
+	ValidTokensTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      kubeServiceAccountSubsystem,
+			Name:           "valid_tokens_total",
+			Help:           "Cumulative valid projected service account tokens used",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+)
+
+var registerMetricsOnce sync.Once
+
+// RegisterMetrics registers kube-proxy metrics.
+func RegisterMetrics() {
+	registerMetricsOnce.Do(func() {
+		legacyregistry.MustRegister(LegacyTokensTotal)
```

Comment:
Why is this using legacyregistry rather than registry? cc @logicalhan @RainbowMango

Reply:
I was referencing other metrics packages in K8S repo. If there is example where we used the registry package please kindly point it to me.

Reply:
I think you should just be able to swap this out for the new registry with no issues? MustRegister is defined: https://github.com/kubernetes/kubernetes/blob/5668dbdc6e60f54d45f2022fddf8a92359cdcac5/staging/src/k8s.io/component-base/metrics/registry.go#L156

Reply:
We should be using the legacyregistry at the moment. It's the same thing as the global prometheus registry. We named it legacyregistry because we intend to move off of it at some point and this makes it easy for us to find it. We have a pretty old issue tracking this somewhere I believe..

Reply:
Link? would love to have that publicly documented :)